pkg/splunk/client/azure_blob_rest_service.go

```diff
@@ -0,0 +1,159 @@
+package client
+
+import (
+	"context"
+)
+
+// Get the list of apps from a given
+// 1. storageaccount URL
+// 2. container name (container in azure is same as bucket in aws)
+// 3. path with the container name
+// 4. Max number of apps to return in the call
+// 5. Config object
+// 	  Timeout - time in ms that a http rest call should wait before terminating
+//    MaxRetries - number of retries for the rest call if it fails for first time.
+//    http transport related parameters
+// Output:
+//  ListBlobOutput
+//     RemoteDataClientResponse
+//     ResponseNextMarker  if total apps in the bucket were more that max number of apps requested
+
+// Download a given app with it's full path
+// 1. URL of the app to be downloaded with full path including app name
+// 2. Max number of bytes to return
+// 3. Starting byte
+// 5. Config object
+// 	  Timeout - time in ms that a http rest call should wait before terminating
+//    MaxRetries - number of retries for the rest call if it fails for first time.
+//    http transport related parameters
+// Output:
+//  DownloadBlobOutput
+//    Response Body
+//    Response Code
+//    Total Number of bytes returned
+
+type azureRestService interface {
+	NewSession(ctx context.Context, appAzureBlobEndPoint string, storageAccountName string, secretAccessKey string, config Config)
+	ListApps(ctx context.Context, bucketName string, opts ListBlobsOptions)
+	DownloadApp(ctx context.Context, bucketName string, remoteFile string, localFile string, opts DownloadOptions)
+}
+
+// NewSession ... : TODO: placeholder for next sprint
+func NewSession(ctx context.Context, appAzureBlobEndPoint string, storageAccountName string, secretAccessKey string, config Config) {
+
+}
+
+// ListApps ... TODO: placeholder for next sprint
+func ListApps(ctx context.Context, bucketName string, opts ListBlobsOptions) {
+}
+
+// DownloadApp ... TODO: placeholder for next sprint
+func DownloadApp(ctx context.Context, bucketName string, remoteFile string, localFile string, opts DownloadOptions) {
+
+}
+
+// func (azureblobClient *AzureBlobClient) DownloadBlob(ctx context.Context, downloadRequest RemoteDataDownloadRequest) (bool, error) {
```

Comment:
nit: remove it for now?

Reply:
I plan to have azure_blob_rest_service.go  refactored in the next sprint (while doing list story). So this whole file may be removed .